core/src/test/java/org/elasticsearch/test/ESBlobStoreContainerTestCase.java

```diff
@@ -111,5 +111,24 @@ public void testMoveAndList() throws IOException {
         }
     }
 
+    public void testOverwriteFails() throws IOException {
+        try (final BlobStore store = newBlobStore()) {
+            final String blobName = "foobar";
+            final BlobContainer container = store.blobContainer(new BlobPath());
+            byte[] data = randomBytes(randomIntBetween(10, scaledRandomIntBetween(1024, 1 << 16)));
+            final BytesArray bytesArray = new BytesArray(data);
+            container.writeBlob(blobName, bytesArray);
+            // should not be able to write to the same blob again
+            try {
+                container.writeBlob(blobName, bytesArray);
+                fail("Cannot overwrite existing blob");
+            } catch (AssertionError e) {
+                // we want to come here
+            }
```

Comment:
(there seems to be a consensus around using expectThrows rather than this try/fail/catch/assert pattern for new tests)